c2aenum/enum_loader.py

```diff
@@ -3,89 +3,74 @@
 
 import os
 import re
+import glob
 
 
 class C2aEnum:
     def __init__(self, c2a_src_path, encoding):
-        self.path = c2a_src_path
+        self.path = c2a_src_path.replace("//", "/")
+        search_path = (self.path + "/**").replace("//", "/")
         self.encoding = encoding
 
-        self._load_bc()
-        self._load_tlm_code()
-        self._load_cmd_code()
-        self._load_app()
-        self._load_anomaly()
-        self._load_el_core_group()
-        self._load_el_group()
-        self._load_eh_rule()
-        self._load_mode()
-        self._load_exec_sts()
-        self._load_tlcd_id()
-
-    def _load_bc(self):
-        self._load_enum_from_file("/src_user/TlmCmd/block_command_definitions.h", "BC_")
-
-    def _load_tlm_code(self):
-        self._load_enum_from_file("/src_user/TlmCmd/telemetry_definitions.h", "Tlm_CODE_")
-
-    def _load_cmd_code(self):
-        self._load_enum_from_file("/src_user/TlmCmd/command_definitions.h", "Cmd_CODE_")
-
-    def _load_app(self):
-        pass
-
-    def _load_anomaly(self):
-        pass
-
-    def _load_el_core_group(self):
-        self._load_enum_from_file("/src_core/System/EventManager/event_logger.h", "EL_CORE_GROUP_")
-
-    def _load_el_group(self):
-        self._load_enum_from_file("/src_user/Settings/System/event_logger_group.h", "EL_GROUP_")
-
-    def _load_eh_rule(self):
-        self._load_enum_from_file(
-            "/src_user/Settings/System/EventHandlerRules/event_handler_rules.h", "EH_RULE_"
-        )
-
-    def _load_mode(self):
-        pass
-
-    def _load_exec_sts(self):
-        self._load_enum_from_file("/src_core/TlmCmd/common_cmd_packet.h", "CCP_EXEC_")
-
-    def _load_tlcd_id(self):
-        self._load_enum_from_file(
-            "/src_core/Applications/timeline_command_dispatcher.h", "TLCD_ID_"
-        )
-
-    def _load_enum_from_file(self, path, prefix):
-        path = self.path + path
-        p_with_id = re.compile(r"^  ({}\w+) += +(\w+)".format(prefix))
-        p_without_id = re.compile(r"^  ({}\w+)".format(prefix))
-
+        p_file_ext = re.compile(r".*\.(c|cc|cpp|h|hh|hpp)$")
+        files_tmp = [
+            file_path.replace("\\", "/")
+            for file_path in glob.glob(search_path, recursive=True)
+            if p_file_ext.fullmatch(file_path)
+        ]
+
+        files = [
+            file_path
+            for file_path in files_tmp
+            if "Examples" not in file_path.replace(self.path, "")
```

Comment:
[IMO]

```suggestion
            if "src_core/Examples" not in file_path.replace(self.path, "")
```

のほうが安全？